snapshots/lcow/lcow.go

```diff
@@ -232,6 +232,16 @@ func (s *snapshotter) Remove(ctx context.Context, key string) error {
 	path := s.getSnapshotDir(id)
 	renamed := s.getSnapshotDir("rm-" + id)
 	if err := os.Rename(path, renamed); err != nil && !os.IsNotExist(err) {
+		// Sometimes if there are some open handles to the files (especially VHD)
+		// inside the snapshot directory the rename call will return "access
+		// denied" or "file is being used by another process" errors.  Just
+		// returning that error causes the entire snapshot garbage collection
+		// operation to fail. To avoid that we return failed pre-condition error
+		// here so that snapshot garbage collection can continue and can cleanup
+		// other snapshots.
+		if strings.Contains(err.Error(), "being used by another process") || os.IsPermission(err) {
```

Comment:
Can this be rewritten as something like `errors.Is(err, windows.ERROR_SHARING_VIOLATION)` instead? That seems cleaner than checking against the string value.

Reply:
Actually I am wondering if we should just return `ErrFailedPrecondition` for all failures. That way snapshotter cleanup operation will still continue and cleanup as many snapshots as it can. 
Do you have any opinions on this @kevpar, @dmcgowan, @samuelkarp ?

Reply:
ping @kevpar, @dmcgowan, @samuelkarp

Reply:
> Actually I am wondering if we should just return `ErrFailedPrecondition` for all failures. That way snapshotter cleanup operation will still continue and cleanup as many snapshots as it can.

What are the other failure scenarios that you're thinking about here?

Reply:
I don't have any specific failures in mind. I was thinking no matter what error it is, it would be always beneficial to just continue with snapshotter cleanup than aborting it.